src/Jenssegers/Mongodb/Connection.php

```diff
@@ -277,4 +291,78 @@ public function __call($method, $parameters)
     {
         return call_user_func_array([$this->db, $method], $parameters);
     }
+
+    /**
+     * create a session and start a transaction in session
+     *
+     * In version 4.0, MongoDB supports multi-document transactions on replica sets.
+     * In version 4.2, MongoDB introduces distributed transactions, which adds support for multi-document transactions on sharded clusters and incorporates the existing support for multi-document transactions on replica sets.
+     * To use transactions on MongoDB 4.2 deployments(replica sets and sharded clusters), clients must use MongoDB drivers updated for MongoDB 4.2.
+     *
+     * @see https://docs.mongodb.com/manual/core/transactions/
+     * @return void
+     */
+    public function beginTransaction()
+    {
+        $this->session_key = uniqid();
+        $this->sessions[$this->session_key] = $this->connection->startSession();
```

Comment:
Given that the session from the connection instance is applied automatically to all operations sent through the query builder, I'd refrain from starting multiple sessions. Looking at the [transaction logic used for PDO](https://github.com/illuminate/database/blob/bd88721cabecee43c9ebc4398a9fcedea7f22092/Concerns/ManagesTransactions.php#L134), the first call to `beginTransaction` will start a new transaction, while subsequent only create save points (or no-op if those are not supported). I generally believe this to be a better alternative that should be used instead.